pkg/envoy/lds.go

```diff
@@ -351,12 +352,19 @@ func findMatchingFilterChainNames(envoyConfig *Config, expectedListenerName stri
 				return ErrUnmarshalingListener
 			}
 			for _, listenerFilter := range listener.FilterChains {
+				log.Error().Msg(listenerFilter.Name)
 				// Check filter chain name
 				actualFilterChainNames = append(actualFilterChainNames, listenerFilter.Name)
-				listenerFilter.GetFilterChainMatch()
-				if found, exists := possibleFilterChainNames[listenerFilter.Name]; exists && !found {
-					// Found expected filter chain
-					possibleFilterChainNames[listenerFilter.Name] = true
+				for expectedFilterChainName := range possibleFilterChainNames {
+					// For osm pre v0.9, the listenerFilter.Name does not have the port number appended to it.
+					// For osm v0.10 onwards, the listenerFilter.Name has the port number appended to it and may also
+					// have the traffic type appended to it.
+					// Examples for listenerFilter.Name v0.10 onwards:
+					//				inbound-mesh-http-filter-chain:bookstore/bookstore-v1:14001
+					//				outbound-mesh-http-filter-chain:bookstore/bookstore-v1_14001_http
```

Comment:
^Let's not make this update now. However, looping through all possible filter chains for each listener filter chain found is inefficient. This can be addressed in a followup PR

Reply:
agreed @jaellio